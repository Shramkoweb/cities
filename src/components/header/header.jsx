import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {getAuthorizationStatus, getUserData} from "../../reducer/user/selector";
import {REQUEST_URL} from "../../constants";

// TODO Адаптер для данных пользователя

const Header = ({isAuthorizationRequired, userData}) => {
  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <a className="header__logo-link" href="main.html">
              <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41"/>
            </a>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                <a className="header__nav-link header__nav-link--profile" href="#">
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                    style={isAuthorizationRequired ? {} : {backgroundImage: `url(${REQUEST_URL.BASE}${userData.avatar})`}}
                  >
                  </div>
                  <span className="header__user-name user__name">
                    {isAuthorizationRequired ? `Sign In` : userData.email}
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  isAuthorizationRequired: PropTypes.bool.isRequired,
  userData: PropTypes.shape({
    avatar: PropTypes.string,
    email: PropTypes.string,
    id: PropTypes.number,
    isPro: PropTypes.bool,
    name: PropTypes.string,
  }),
};


const mapStateToProps = (state) => ({
  isAuthorizationRequired: getAuthorizationStatus(state),
  userData: getUserData(state)
});


export {Header};
export default connect(mapStateToProps)(Header);
