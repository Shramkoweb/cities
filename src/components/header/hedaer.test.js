import React from "react";
import renderer from "react-test-renderer";
import {Header} from "./header";

it(`Header component render correct`, () => {
  const userMock = {
    avatar: `static/avatar/10.jpg`,
    email: `Shramko.web@yahoo.com`,
    id: 13,
    isPro: false,
    name: `Shramko.web`,
  };

  const header = renderer
    .create(<Header userData={userMock} isAuthorizationRequired={false}/>)
    .toJSON();

  expect(header).toMatchSnapshot();
});
