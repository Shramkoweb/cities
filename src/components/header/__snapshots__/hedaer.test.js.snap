// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render correct 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
          href="main.html"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="#"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
                style={
                  Object {
                    "backgroundImage": "url(https://htmlacademy-react-2.appspot.com/six-citiesstatic/avatar/10.jpg)",
                  }
                }
              />
              <span
                className="header__user-name user__name"
              >
                Shramko.web@yahoo.com
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
